src/rgw/rgw_rest_s3.cc

```diff
@@ -1304,7 +1304,7 @@ struct ReplicationConfiguration {
       *enabled = (status == "Enabled");
 
       pipe->params.mode = rgw_sync_pipe_params::Mode::MODE_USER;
-      pipe->params.user = user_id.to_str();
+      pipe->params.user = to_string(s->owner.id);
```

Comment:
@cbodley - I’ve identified an issue here. We’re always passing `s->owner.id` in `rgwx-uid`, but in the accounting scenario, this might cause confusion because `s->owner.id` is the account ID, not the user ID. For scenarios like this, we’re more interested in the user, so we can later load the user and perform authz based on the user, not the account (for this scenario even the direct request is the problem - as `s->owner.id` is still the account id).

What do you think? Should we pass the user ID instead of the owner ID in this case, or take another approach?

Reply:
i don't really understand the purpose of this code, but are we sure it actually needs to run on the master zone?

Reply:
I guess so, as this should be part of the bucket's metadata and be synced to all zones in case it is pointing to multiple buckets for example.

Reply:
In your commit here (https://github.com/ceph/ceph/commit/8d50f6103ff481a6ef4bf7c894e1d4329eefcf49) you specified that the header should be only used for ownership assignment, so now I'm confused with this case and the case here I describe for source object authz (https://github.com/ceph/ceph/pull/60685).

Reply:
sorry, this sync policy stuff has always confused me. i see that this code is responsible for initializing `RGWBucketInfo::sync_policy`, which does need to happen on the master 👍 

what i don't understand is why the _requesting user_ is ending up in this sync policy. why would that be anything other than the bucket's owner?

Reply:
Yeah, in AWS the story is to set STS Role rather than a user... so I guess we can for the time being point to the bucket's owner rather than a requesting user. But still with the bucket owner, in the accounting scenario whom should it be for further authz in data sync (https://github.com/ceph/ceph/pull/60685)? I guess bucket owners are still `ACLOwner` not `rgw_user`.

Reply:
yeah, i don't know. i didn't attempt to define an interaction between accounts and sync policies. for now, we might reject the request if the 'owner' is an account?

Reply:
yeah, that can be a solution, we can return `ERR_NOT_IMPLEMENTED` perhaps if the `s->owner` was an account rather than a user. So later we can fix this.